python/src/scipp/_structured.py

```diff
@@ -23,11 +23,34 @@ def is_structured(obj):
 
 def _fields(obj):
     class Fields():
-        def __init__(self, var):
+        def __init__(self, keys, var):
             self._var = var
+            self._keys = keys
+
+        def __contains__(self, key):
+            return key in self._keys
+
+        def __iter__(self):
+            return self._keys.__iter__()
+
+        def __getitem__(self, key):
+            return _get_elements(self._var, key)
+
+        def __setitem__(self, key, x):
+            return _set_elements(self._var, key, x)
+
+        def keys(self):
+            return self._keys
+
+        def values(self):
+            return [self[key] for key in self]
+
+        def items(self):
+            return {key: self[key] for key in self}
```

Comment:
Make them lazy:
```.py
def values(self):
    yield from (self[key] for key in self)

def items(self):
    yield from ((key, self[key]) for key in self)
```
This also changes `items` to return an iterator over tuples instead of a dict. Currently one needs to call `fields.items().items()`.

Reply:
Should it also be `yield from self._keys` in `keys()`?

Reply:
Good point, yes

Reply:
As far as I can tell `keys()` is actually supposed to rerturn a list, so I didn't change it?

Reply:
Not really:
```.py
>>> d={'a':2, 'b':3}
>>> d.keys()
dict_keys(['a', 'b'])
>>> d.keys()[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_keys' object is not subscriptable
>>> dir(d.keys())
['__and__', '__class__', '__contains__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__ne__', '__new__', '__or__', '__rand__', '__reduce__', '__reduce_ex__', '__repr__', '__reversed__', '__ror__', '__rsub__', '__rxor__', '__setattr__', '__sizeof__', '__str__', '__sub__', '__subclasshook__', '__xor__', 'isdisjoint']
```
I don't think using `list` is a serious problem but it does behave differently.